src/hooks/use-document-sync-to-firebase.ts

```diff
@@ -64,7 +64,7 @@ export function useDocumentSyncToFirebase(
 
   // sync properties for personal and learning log documents
   useSyncMstNodeToFirebase({
-    firebase, model: document.properties, path: typedMetadata,
+    firebase, model: document.properties, path: `${metadata}/properties`,
```

Comment:
I think if you just add `ProblemDocument` to the list of enabled types below you'd fix the issue with problem document versions. It's simply that properties changes aren't being synchronized back to firebase at this point.